### PR TITLE
BLS multisig without embedded rogue key prevention

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
When relying on KOSK there is no additional value in using the embedded rogue key prevention.
This PR creates a version of multisigner (the low level part) that relies on KOSK for the rogue key prevention, and by doing so has an optimized aggregation and verification (of the aggregated sig).

This multisigner should be used only if KOSK is ensured for all signers.

### CODE REVIEW

File: signing/mcl/multisig/blsKOSK_test.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+		err = llSig.VerifyAggregatedSig(pubKeys[0].Suite(), pubKeys, nil, msg)
+		require.Equal(t, crypto.ErrNilSignature, err)
+
+		err = llSig.VerifyAggregatedSig(pubKeys[0].Suite(), pubKeys, []byte{}, msg)
+		require.Equal(t, crypto.ErrNilSignature, err)
+	})
+	t.Run("invalid aggregated sig should err", func(t *testing.T) {
+		err = llSig.VerifyAggregatedSig(pubKeys[0].Suite(), pubKeys, sigShares[0], msg)
+		require.Equal(t, crypto.ErrAggSigNotValid, err)
+
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply signing/mcl/multisig/blsKOSK_test.go:205]: done

File: signing/mcl/multisig/blsKOSK.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing/mcl"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing/mcl/singlesig"
+	"github.com/herumi/bls-go-binary/bls"
+)
+
+var _ crypto.LowLevelSignerBLS = (*BlsMultiSignerKOSK)(nil)
+
+// BlsMultiSignerKOSK provides an implements of the crypto.LowLevelSignerBLS interface
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply signing/mcl/multisig/blsKOSK.go:13]: done

File: signing/mcl/multisig/blsKOSK_test.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+		}
+
+		err := lls.VerifySigShare(pk, msg, sig)
+		require.Equal(t, crypto.ErrInvalidPublicKey, err)
+	})
+	t.Run("nil msg should err", func(t *testing.T) {
+		err := lls.VerifySigShare(pk, nil, sig)
+		require.Equal(t, crypto.ErrNilMessage, err)
+	})
+	t.Run("nil msg should err", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply signing/mcl/multisig/blsKOSK_test.go:112]: done

**Review by raduchis — APPROVED** [review]

**Review by ssd04 — DISMISSED** [review]

File: signing/mcl/multisig/common.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
+package multisig
```
Comment (ccorcoveanu) [inline signing/mcl/multisig/common.go:1]: Can we group these function in one (if just one, that would mean we just rename this one) or two files with related funcs? So we won't add this `common.go` file. It's an invitation to pile other unrelated stuff here

**Review by raduchis — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Review by ccorcoveanu — APPROVED** [review]
